internal/gps/solver.go

```diff
@@ -186,17 +190,21 @@ func (params SolveParameters) toRootdata() (rootdata, error) {
 	if params.Manifest == nil {
 		params.Manifest = simpleRootManifest{}
 	}
+	if params.stdLibFn == nil {
+		params.stdLibFn = paths.IsStandardImportPath
+	}
 
 	rd := rootdata{
-		ig:      params.Manifest.IgnoredPackages(),
-		req:     params.Manifest.RequiredPackages(),
-		ovr:     params.Manifest.Overrides(),
-		rpt:     params.RootPackageTree.Copy(),
-		chng:    make(map[ProjectRoot]struct{}),
-		rlm:     make(map[ProjectRoot]LockedProject),
-		chngall: params.ChangeAll,
-		dir:     params.RootDir,
-		an:      params.ProjectAnalyzer,
+		ig:       params.Manifest.IgnoredPackages(),
+		req:      params.Manifest.RequiredPackages(),
+		ovr:      params.Manifest.Overrides(),
+		rpt:      params.RootPackageTree.Copy(),
+		chng:     make(map[ProjectRoot]struct{}),
+		rlm:      make(map[ProjectRoot]LockedProject),
+		chngall:  params.ChangeAll,
+		dir:      params.RootDir,
+		an:       params.ProjectAnalyzer,
+		stdLibFn: params.stdLibFn,
```

Comment:
This isn't really "rootdata" - that struct is generally supposed to hold information that describes the root project itself. Let's put it directly on the `solver`, instead.

(I realize, when looking again at `rootdata`, that I got sloppy and violated this definition by putting the `ProjectAnalyzer` in there - but let's not worsen the problem)